edx_rest_api_client/auth.py

```diff
@@ -48,3 +48,16 @@ def __call__(self, r):
         """ Update the request headers. """
         r.headers['Authorization'] = 'Bearer {}'.format(self.token)
         return r
+
+
+class OidcIdTokenAuth(AuthBase):
```

Comment:
@clintonb and @rlucioni your comments make sense, however I have what I think is a cleaner counter-proposal to address the concerns:

1) take what @zubair-arbi implemented as `OidcIdTokenAuth` and rename it to `JwtAuth` (replacing the current one)
2) remove the internal logic from the current `JwtAuth` implementation ([this](https://github.com/edx/edx-rest-api-client/blob/zub/add-open-id-auth/edx_rest_api_client/auth.py#L21-L34)) that creates and signs the JWT, and put it instead in a helper method on `EdxRestApiClient`.
3) In `EdxRestApiClient.__init__`, pass the id_token to `JwtAuth` if we got one, and if not, generate the JWT using `signing_key`, `username` etc in the helper method.

After this change, the `JwtAuth` and `BearerAuth` classes will do essentially the same thing, just pass along an already-existing token with the correct Authorization header.

What do you think?